src/components/data/reports/CsvReport.js

```diff
@@ -1,49 +1,39 @@
 import { get } from 'lodash';
 import { exportCsv } from '@folio/stripes/util';
 import moment from 'moment';
+import reportColumns from './reportColumns';
 
-const columns = [
-  'borrower.name',
-  'borrower.barcode',
-  'borrowerId',
-  'dueDate',
-  'loanDate',
-  'loanPolicy.name',
-  'loanPolicyId',
-  'loanId',
-  'feeFine',
-  'item.title',
-  'item.materialType.name',
-  'item.status.name',
-  'item.barcode',
-  'item.callNumberComponents.prefix',
-  'item.callNumberComponents.callNumber',
-  'item.callNumberComponents.suffix',
-  'item.volume',
-  'item.enumeration',
-  'item.chronology',
-  'item.copyNumber',
-  'item.contributors',
-  'item.location.name',
-  'item.instanceId',
-  'item.holdingsRecordId',
-  'itemId',
-];
 
-class OverdueLoanReport {
+
+class CsvReport {
   constructor(options) {
     const { formatMessage } = options;
+    this.formatMessage = formatMessage;
+  }
+
+  setUp(type) {
+    const overDueDate = moment().tz('UTC').format();
+    let columns;
+    this.queryString = '';
+    if (type === 'overdue') {
```

Comment:
@K-Felk could `type` be used as a `key` in `reportColumns`? That way you could skip the if/else statements.

For example:

````js
const columns = reportColumns[type];
````

You could also extract `queryString` in the same fashion maybe as a function since you have to use `overDueDate` as a dynamic param but something similar to this should work:

````js
this.queryString = queryStrings[type]();
```

Reply:
that is simpler.  I'll refactor.